common/functions/src/scalars/dates/number_function.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::marker::PhantomData;
+
+use common_datavalues::arrays::DFUInt32Array;
+use common_datavalues::chrono::DateTime;
+use common_datavalues::chrono::Datelike;
+use common_datavalues::chrono::TimeZone;
+use common_datavalues::chrono::Utc;
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::Function;
+
+#[derive(Clone, Debug)]
+pub struct NumberFunction<T> {
+    display_name: String,
+    t: PhantomData<T>,
+}
+
+pub trait NumberResultFunction {
+    fn execute(_value: DateTime<Utc>) -> u32;
+}
+
+#[derive(Clone)]
+pub struct ToYYYYMM;
+
+impl NumberResultFunction for ToYYYYMM {
+    fn execute(value: DateTime<Utc>) -> u32 {
+        value.year() as u32 * 100 + value.month()
+    }
+}
+
+impl<T> NumberFunction<T>
+where T: NumberResultFunction + Clone + Sync + Send + 'static
+{
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(NumberFunction::<T> {
+            display_name: display_name.to_string(),
+            t: PhantomData,
+        }))
+    }
+}
+
+impl<T> Function for NumberFunction<T>
+where T: NumberResultFunction + Clone + Sync + Send + 'static
+{
+    fn name(&self) -> &str {
+        self.display_name.as_str()
+    }
+
+    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
+        Ok(DataType::UInt32)
+    }
+
+    fn num_arguments(&self) -> usize {
+        1
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, _input_rows: usize) -> Result<DataColumn> {
+        if columns.is_empty() {
+            return Result::Err(ErrorCode::UnknownFunction(
+                "Number of arguments for function toYYYYMM doesn't match: passed 0, should be 1 or 2")
+            );
+        }
+
+        let data_type = columns[0].data_type();
+        let number_vec = match data_type {
+            DataType::Date16 | DataType::Date32 => {
+                let number_vec_result: Vec<Option<u32>> = columns[0]
+                    .column()
+                    .to_values()?
```

Comment:
`to_values` has poor performance than native array iter.


How about using `let col = columns[0].u16()?` and `let col = columns[0].u32()?`. 
Then we can have `col.iter().map(...)`

Reply:
Is the u16() method in SeriesTrait?

Reply:
Yes